pkg/provider/hpa.go

```diff
@@ -162,15 +165,19 @@ func (p *HPAProvider) updateHPAs() error {
 					interval = p.collectorInterval
 				}
 
-				collector, err := p.collectorFactory.NewCollector(&hpa, config, interval)
+				c, err := p.collectorFactory.NewCollector(&hpa, config, interval)
 				if err != nil {
-					p.recorder.Eventf(&hpa, apiv1.EventTypeWarning, "CreateNewMetricsCollector", "Failed to create new metrics collector: %v", err)
+
+					if errors.Is(err, &collector.PluginNotFoundError{}) && !p.disregardIncompatibleHPAs {
```

Comment:
We should be able to log errors which are NOT `PluginNotFoundError`. Currently those are not reported.

Reply:
Indeed, the latest commit addresses this. Even did a truth table on paper to make sure the expression was correct.

Reply:
Doesn't seem right. Now we would only log if `disregardIncompatibleHPAs` is enabled.

We should log any error when `disregardIncompatibleHPAs=false` and log all errors which is not `PluginNotFoundError` when `disregardIncompatibleHPAs=true`.

Reply:
Ok, let me reproduce my truth table here:

Log = !(errors.Is(err, &collector.PluginNotFoundError{}) && p.disregardIncompatibleHPAs)

| Log | is PluginNotFoundError | disregardIncompatibleHPAs = true |
|-|-|-|
|F|T|T|
|T|T|F|
|T|F|T|
|T|F|F|

Reply:
OK, I missed the wrapping `!()`. It seems a bit hard to read this way, but as it's correct I can live with it :)

Reply:
nice! 😄